src/pages/pupils/beta/programmes/[programmeSlug]/units.tsx

```diff
@@ -0,0 +1,163 @@
+import { GetStaticProps, GetStaticPropsResult } from "next";
+import Link from "next/link";
+
+import { UnitListingBrowseData } from "@/node-lib/curriculum-api-2023/queries/pupilUnitListing/pupilUnitListing.schema";
+import { URLParams } from "@/pages/teachers/programmes/[programmeSlug]/units";
+import getPageProps from "@/node-lib/getPageProps";
+import curriculumApi2023 from "@/node-lib/curriculum-api-2023";
+import { resolveOakHref } from "@/common-lib/urls";
+
+export { getStaticPaths } from "@/pages/teachers/programmes/[programmeSlug]/units";
+
+type UnitListingPageProps = {
+  curriculumData: UnitListingBrowseData;
+};
+
+const PupilUnitListingPage = ({ curriculumData }: UnitListingPageProps) => {
+  console.log("curriculumData", curriculumData);
+
+  const orderedCurriculumData = curriculumData.sort((a, b) => {
+    const aUnitOrder = a.supplementaryData.unitOrder;
+    const bUnitOrder = b.supplementaryData.unitOrder;
+    return aUnitOrder - bUnitOrder;
+  });
+
+  const subject = curriculumData[0]?.programmeFields?.subject;
+  const yearDescription = curriculumData[0]?.programmeFields?.yearDescription;
+  const tier = curriculumData[0]?.programmeFields?.tier || null;
+  const examboard = curriculumData[0]?.programmeFields?.examboard;
+  const yearSlug = curriculumData[0]?.programmeFields?.yearSlug || "";
+  const examboardSlug =
+    curriculumData[0]?.programmeFields?.examboardSlug || null;
+  const subjectSlug = curriculumData[0]?.programmeFields?.subjectSlug || "";
+
+  type BackPages =
+    | "tier-with-examboard"
+    | "tier-without-examboard"
+    | "examboard"
+    | "subject";
+
+  function pickPreviousPage(): BackPages {
+    const hasTier = tier !== null;
+    const hasExamboard = examboard !== null;
```

Comment:
this will cause problems as examboard can be undefined. A falsy comparison would be better eg.
```hasExamboard = !!examboard```